src/Tickets/Admin/Upsell.php

```diff
@@ -221,12 +221,23 @@ public function maybe_show_paystack_promo( $settings ) {
 	 * Filters the default Ticket type description in the context of Events part of a Series.
 	 *
 	 * @since 5.8.0
+	 * @since TBD   Add logic to bail in scenarios when upsell should not show.
 	 *
 	 * @param string   $file     Complete path to include the PHP File.
 	 * @param string[] $name     Template name.
 	 * @param Template $template Current instance of the Tribe__Template.
 	 */
 	public function render_ticket_type_upsell_notice( string $file, array $name, Template $template ): void {
+		// Check if post type is an event.
+		if ( ! function_exists( 'tribe_is_event' ) || ! tribe_is_event() ) {
+			return;
+		}
+
+		// If not within the admin area, then bail.
+		if ( ! is_admin() ) {
+			return;
+		}
```

Comment:
I would use an action like `did_action( 'action_that_only_happens_in_pro' )`, that tends to be a safer bet.